### DISCUSSION TITLE
Swimlane support for Kanban layout

### DISCUSSION THREAD

**Discussion by jpfcu** [directus/directus#18859] [Feature Requests]
### Summary

Add swimlanes to the Kanban layout in order to group items by another level besides the columns. 

### Basic Example

<img width="100%" alt="swimlanes_jira" src="https://github.com/directus/directus/assets/95651583/702a9710-c536-4e3f-9d9c-f7f90198b01a">
<img width="100%" alt="swimlanes_trello" src="https://github.com/directus/directus/assets/95651583/4f4f1255-b7bd-43f9-bf0f-56bc89f2454f">

### Motivation

Swimlanes provide an effective visual organization method in Kanban boards. By introducing swimlanes to directus, users will be able to categorize and group tasks or items based on specific criteria, such as teams, projects, or priority levels. This feature will enhance the flexibility and usability of the Kanban layout, making it easier for users to manage and track their workflows.

### Detailed Design

The swimlanes feature should allow users to create and customize swimlanes within the Kanban layout. Users should be able to define the criteria for swimlane creation, such as custom fields or labels. 

The swimlanes should be visually distinct and display relevant information for each lane, such as lane titles and associated tasks or items. 

Additionally, the swimlanes should support drag-and-drop functionality to easily move tasks or items between lanes.

### Must Have:
* Ability to create custom swimlanes within the Kanban layout.
* Visual distinction and clear titles for each swimlane.
* Drag-and-drop functionality to move tasks or items between swimlanes.

### Should Have:
* Ability to collapse or expand swimlanes for better visibility.
* Filtering options to view specific swimlanes or combinations of swimlanes.

### Could Have:
* Ability to toggle visibility of empty swimlanes
* Ability to _show on top_, _show on bottom_ or _hide_ items without a swimlane assignment, e.g. for _null_ values of corresponding field.
* Swimlane-level metrics or statistics, such as task counts or completion rates.
* Configurable swimlane colors or styles for improved customization.

### Drawbacks

- Increased complexity of the Kanban layout may affect performance or usability for users with large datasets or low-spec devices.
- Additional grouping level may increase complexity in validating permissions for drag-and-drop actions.
- The learning curve for users familiar with the existing Kanban layout may be slightly steeper.
- The configuration interface for the kanban layout would be become more complex and may need a structural update. 

### Alternatives

Instead of swimlanes, users can currently rely on labels or filtering within the Kanban layout to categorize and group tasks. However, swimlanes provide a more intuitive and visually distinct representation of groups, making it easier to identify and manage tasks within specific categories.

### Adoption Strategy

* no breaking change expected, as grouping in swimlanes shall be optional

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Comment by cdierkens** [comment]


**Reply by br41nslug** [reply]
That link does not seem related to a Directus repository 🤔 And no this is still an open feature request that has not been implemented yet, we dont have enterprise exclusive features (anymore)

**Reply by cdierkens** [reply]
Sorry for the noise!
